crm_timesheet/models/crm_phonecall.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+# License AGPL-3 - See LICENSE file on root folder for details
+##############################################################################
+
+from openerp import models, fields, api
+from datetime import datetime
+
+import logging
+from pprint import pformat
+_logger = logging.getLogger(__name__)
+
+
+class CrmPhonecall(models.Model):
+    _inherit = 'crm.phonecall'
+
+    analytic_account = fields.Many2one(
+        comodel_name='account.analytic.account')
+    timesheets = fields.One2many(comodel_name='hr.analytic.timesheet',
+                                 inverse_name='phonecall')
+
+    def _get_timesheet_vals(self, vals):
+        date = vals.get('date')
+        user_id = vals.get('user_id')
+        name = vals.get('name')
+        account_id = vals.get('analytic_account')
+        unit_amount = vals.get('duration')
+        res = {
+            'date': date[:10] if date else self.date[:10],
+            'user_id': user_id if user_id else self.user_id.id,
+            'name': name if name else self.name,
```

Comment:
You don't need to put this expression. It's enough with `'name': name or self.name,`. This is because Python return the first non false expression when making boolean operations. But you can do even better putting directly: `'name': vals.get('name', self.name)`, because get operator allows to pass a value to be given as fallback value.